istioctl/cmd/tag.go

```diff
@@ -0,0 +1,400 @@
+	cmd.PersistentFlags().StringVarP(&revision, "revision", "r", "", "revision to point tag to")
+	_ = cmd.MarkPersistentFlagRequired("revision")
+
+	return cmd
+}
+
+func tagListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List existing revision tags and their corresponding revisions",
+		Example: "istioctl x tag list",
+		Aliases: []string{"show"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return fmt.Errorf("tag list command does not accept arguments")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+			return listTags(context.Background(), client.Kube(), cmd.OutOrStdout())
+		},
+	}
+
+	return cmd
+}
+
+func tagRemoveCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "remove",
+		Short:   "Remove an existing revision tag",
+		Example: "istioctl x tag remove prod",
+		Aliases: []string{"delete"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("must provide a tag for removal")
+			}
+			if len(args) > 1 {
+				return fmt.Errorf("must provide a single tag for removal")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+
+			return removeTag(context.Background(), client.Kube(), args[0], skipConfirmation, cmd.OutOrStdout())
+		},
+	}
+
+	return cmd
+}
+
+// setTag creates or modifies a revision tag.
+func setTag(ctx context.Context, kubeClient kubernetes.Interface, tag, revision string, w io.Writer) error {
+	revWebhooks, err := getWebhooksWithRevision(ctx, kubeClient, revision)
+	if err != nil {
+		return err
+	}
+	if len(revWebhooks) == 0 {
+		return fmt.Errorf("cannot find webhook with revision %q", revision)
+	}
+	if len(revWebhooks) > 1 {
+		return fmt.Errorf("found multiple canonical webhooks for revision %q", revision)
+	}
+
+	tagWebhook, err := buildTagWebhookFromCanonical(revWebhooks[0], tag, revision)
+	if err != nil {
+		return err
+	}
+
+	whs, err := getWebhooksWithTag(ctx, kubeClient, tag)
+	if err != nil {
+		return err
+	}
+
+	if len(whs) > 0 && !overwrite {
+		return fmt.Errorf("found an existing revision tag for %q, use the --overwrite flag to overwrite", tag)
+	}
+
+	// create or update webhook depending on whether it exists
+	// note there should be a single webhook for each revision tag
+	kubeWebhookClient := kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations()
+	if len(whs) == 0 {
+		_, err = kubeWebhookClient.Create(ctx, tagWebhook, metav1.CreateOptions{})
+	} else {
+		tagWebhook.ObjectMeta.ResourceVersion = whs[0].ObjectMeta.ResourceVersion
+		_, err = kubeWebhookClient.Update(ctx, tagWebhook, metav1.UpdateOptions{})
+	}
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(w, "Revision tag %s created\n", tag)
```

Comment:
It might be valuable to supply more output here:

Revision tag %q created, referring to Istio control plane revision %q.  To enable injection using this revision tag, use 'kubectl label namespace <namespace> istio.io/rev=%s'"